docs/ref/models/fields.txt

```diff
@@ -426,6 +426,11 @@ precedence when creating instances in Python code. ``db_default`` will still be
 set at the database level and will be used when inserting rows outside of the
 ORM or when adding a new field in a migration.
 
+.. note::
+
+    ``DatabaseDefault()`` class is an internal class which provides a
+    mechanism to allow db_default to store values at database level.
```

Comment:
I don't feel like this covered the confusion which was mentioned in the ticket, maybe something like...
```suggestion
If a field has a ``db_default`` without a :attr:`Field.default` set, unsaved
model instances which have not assigned a value to the field will return a
``DatabaseDefault``` object when accessing the field value. The value is
calculated when the model instance is saved to the database.
```

Reply:
Okay makes sense. Let me add both of the commits and go through the texts again.

Reply:
@sarahboyce Should we add thi

> I don't feel like this covered the confusion which was mentioned in the ticket, maybe something like...
> 
> Although the sentence is horribly long 🤔 but hopefully you see where I'm going with it

Should we put this message inside a note? What do you think?

Reply:
```suggestion
.. admonition:: The field value is a ``DatabaseDefault`` object prior to saving

    Unsaved model instances that have not set a value to a field with a
    ``db_default`` (without a ``default`` set) will return a ``DatabaseDefault``
    object when accessing the field value.

    .. code-block:: shell

        >>> my_model_instance = MyModel()
        >>> my_model_instance.created
        <django.db.models.expressions.DatabaseDefault object at 0x7a5d29cad610>

    The value is calculated when the model instance is saved to the database.
```
Maybe

Reply:
Okay i exactly had this in mind, but thought would become too lengthy. But here being more descriptive will help. Making the changes in the commit.

Reply:
Docs are hard - we might iterate a bit before we're happy :+1:

Reply:
Totally! I get it! But in the latest commit i don't understand why the "Docs / blacken-docs (pull_request) " is failing.